src/WikiService.Validation/Rubric/EditRubricRequestValidator.cs

```diff
@@ -0,0 +1,195 @@
+using LT.DigitalOffice.WikiService.Data.Interfaces;
+using LT.DigitalOffice.WikiService.Models.Db;
+using LT.DigitalOffice.WikiService.Models.Dto.Requests.Rubric;
+using LT.DigitalOffice.WikiService.Validation.Rubric.Interfaces;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.WikiService.Validation.Rubric
+{
+  public class EditRubricRequestValidator : ExtendedEditRequestValidator<Guid, EditRubricRequest>, IEditRubricRequestValidator
+  {
+    private readonly IRubricRepository _rubricRepository;
+
+    private async Task HandleInternalPropertyValidationAsync(
+      Operation<EditRubricRequest> requestedOperation,
+      CustomContext context)
+    {
+      Context = context;
+      RequestedOperation = requestedOperation;
+
+      #region paths
+
+      AddСorrectPaths(
+        new List<string>
+        {
+          nameof(EditRubricRequest.Name),
+          nameof(EditRubricRequest.ParentId),
+          nameof(EditRubricRequest.IsActive)
+        });
+
+      AddСorrectOperations(nameof(EditRubricRequest.Name), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditRubricRequest.ParentId), new List<OperationType> { OperationType.Replace });
+      AddСorrectOperations(nameof(EditRubricRequest.IsActive), new List<OperationType> { OperationType.Replace });
+
+      #endregion
+
+      #region Name
+
+      AddFailureForPropertyIf(
+        nameof(EditRubricRequest.Name),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            x => !string.IsNullOrWhiteSpace(x.value?.ToString()),
+            "Name must not be empty."
+          },
+          {
+            x => x.value.ToString().Trim().Length <= 150,
+            "Name is too long."
+          }
+        }, CascadeMode.Stop);
+
+      #endregion
+
+      #region IsActive
+
+      AddFailureForPropertyIf(
+       nameof(EditRubricRequest.IsActive),
+       x => x == OperationType.Replace,
+       new()
+       {
+         {
+           x => bool.TryParse(x.value?.ToString(), out bool _), "Incorrect rubric is active format"
+         },
+       }, CascadeMode.Stop);
+
+      #endregion
+
+      #region ParentId
+
+      await AddFailureForPropertyIfAsync(
+        nameof(EditRubricRequest.ParentId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          {
+            x => Task.FromResult(
+              string.IsNullOrEmpty(x.value?.ToString())
+              ? true
+              : Guid.TryParse(x.value.ToString(), out Guid result)),
+            "Incorrect format of ParentId."
+          },
+          {
+            async (x) =>
+            {
+              if (x.value?.ToString() is null)
+              {
+                return true;
+              }
+
+              if (!Guid.TryParse(x.value?.ToString(), out Guid parentId))
```

Comment:
you've already checked the value to being not null and in right format (lines 83-89). Is it necessary to check it again?